packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -96,8 +96,8 @@ export class ApolloServerBase {
   public subscriptionsPath?: string;
   public graphqlPath: string = '/graphql';
   public requestOptions: Partial<GraphQLOptions<any>> = Object.create(null);
+  public context?: Context | ContextFunction;
```

Comment:
I can see how this is useful functionality to add to the `createTestClient` utility, but I'm afraid that just changing this previously-private `context` of `ApolloServerBase` (which is sub-classed elsewhere) into a public property, free to be manipulated without guard, is over-exposing the `context`.  

Any ideas what another approach here might look like?